SwissTransferFeatures/UploadProgressView/TransferSessionManager.swift

```diff
@@ -104,58 +92,116 @@ class TransferSessionManager: ObservableObject {
         let remoteUploadFiles = uploadWithRemoteContainer.files.compactMap { $0.remoteUploadFile }
         assert(remoteUploadFiles.count == uploadWithRemoteContainer.files.count, "All files should have a remote upload file")
 
-        for (index, remoteUploadFile) in remoteUploadFiles.enumerated() {
-            let localFile = uploadWithRemoteContainer.files[index]
+        let transferManagerWorker = TransferManagerWorker(overallProgress: overallProgress)
 
-            try await uploadFile(atPath: localFile.localPath, toRemoteFile: remoteUploadFile, uploadUUID: uploadSession.uuid)
-        }
+        try await remoteUploadFiles.enumerated()
+            .map { (uploadWithRemoteContainer.files[$0.offset], $0.element) }
+            .asyncForEach { localFile, remoteUploadFile in
+                try await transferManagerWorker.uploadFile(
+                    atPath: localFile.localPath,
+                    remoteUploadFileUUID: remoteUploadFile.uuid,
+                    uploadUUID: uploadSession.uuid
+                )
+            }
 
         Logger.general.info("Found container: \(container.uuid)")
 
         let transferUUID = try await uploadManager.finishUploadSession(uuid: uploadSession.uuid)
 
         return transferUUID
     }
+}
+
+struct TransferManagerWorker {
```

Comment:
Maybe we could move this struct in a dedicated file ? (I'm not sure about it, up to you)